programs/perpetuals/tests/anchor/basic.ts

```diff
@@ -408,6 +419,80 @@ describe("perpetuals", () => {
     expect(JSON.stringify(oracle)).to.equal(JSON.stringify(oracleExpected));
   });
 
+  it("setCustomOraclePricePermissionless", async () => {
+    await tc.setCustomOraclePricePermissionless(
+      permissionlessOracleUpdateKey,
+      500,
+      tc.custodies[0]
+    );
+
+    let oracle = await tc.program.account.customOracle.fetch(
+      tc.custodies[0].oracleAccount
+    );
+    let oracleExpected = {
+      price: new BN(500000),
+      expo: -3,
+      conf: new BN(0),
+      ema: new BN(500000),
+      publishTime: oracle.publishTime,
+    };
+    expect(JSON.stringify(oracle)).to.equal(JSON.stringify(oracleExpected));
+
+    // Updating the permissionless price oracle with an older publish time should no-op.
+    await tc.setCustomOraclePricePermissionless(
+      permissionlessOracleUpdateKey,
+      400,
+      tc.custodies[0],
+      tc.getTime() - 20
+    );
+    oracle = await tc.program.account.customOracle.fetch(
+      tc.custodies[0].oracleAccount
+    );
+    // Oracle's value is still 500 instead of the attempted 400.
+    expect(JSON.stringify(oracle)).to.equal(JSON.stringify(oracleExpected));
+
+    // after test, set price back to the expected for other test cases.
+    await tc.setCustomOraclePricePermissionless(
```

Comment:
could you add one more test please, I want to make sure setting priority fees and CUs still work. So please build a tx consisting of the following ixs: 1. ed25519 program 2. setComputeUnitLimit 3. SetComputeUnitPrice 4. oracle price update
https://solanacookbook.com/references/basic-transactions.html#how-to-change-compute-budget-fee-priority-for-a-transaction

Reply:
yes!  added a test case for `increaseComputeLimits` above.  confirmed tests pass as expected